api/v1alpha1/cdapmaster_types.go

```diff
@@ -93,6 +93,10 @@ type CDAPMasterSpec struct {
 	Authentication *AuthenticationSpec `json:"authentication,omitempty"`
 	// SecurityContext defines the security context for all pods for all services.
 	SecurityContext *SecurityContext `json:"securityContext,omitempty"`
+	// CustomVolumes defines a list of custom volumes for the service.
+	CustomVolumes []corev1.Volume `json:"customVolumes,omitempty"`
```

Comment:
Does this require yaml to specify an empty array like: 

CustomerVolumes: []

Or it is fine to omit CustomerVolumes in the yaml ?

Reply:
Nit: give some more context about this custom volume in the comment. A sample snippet of how it might be configured would be even better.

Reply:
nit: instead of calling these custom volumes, they are just additional volumes that are mounted on all services, if that is the case, just call them additionalVolumes. These volumes can PD, projected, or any types of volumes.  

"custom" makes it sound like a modified type of volume.

Reply:
Switched to "AdditionalVolumes" and "AdditionalVolumeMounts" and added an additional comment pointing to the Kubernetes documentation on volumes.

Regarding empty lists, it is fine to omit the list entirely; the JSON or YAML deserialization should properly handle the null list case.